dask/multiprocessing.py

```diff
@@ -4,9 +4,27 @@
 from .optimize import fuse, cull
 import multiprocessing
 import dill
+import cloudpickle
 import pickle
 from .async import get_async # TODO: get better get
 from .context import _globals
+from sys import getrecursionlimit, setrecursionlimit
+
+def _dumps(x):
+    try:
+        return cloudpickle.dumps(x, protocol=pickle.HIGHEST_PROTOCOL)
+    except:
+        try:
+            recursionlimit = getrecursionlimit()
+            setrecursionlimit(2*recursionlimit)
+            dumped = cloudpickle.dumps(x, protocol=pickle.HIGHEST_PROTOCOL)
+            setrecursionlimit(recursionlimit)
+            return dumped
+        except:
+            raise
```

Comment:
@mrocklin  I am investigating a bit more about what is going on in the test cases that hit the recursion limit.  For now it looks like this might not be related to the depth of the object we are pickling but rather the depth of the interpreter stack: https://docs.python.org/3.5/library/sys.html#sys.getrecursionlimit .

Running the failing test functions one by one on my laptop (on py27) I cannot reproduce the `recursionlimit` errors.    The errors do not occur on py34